File: test-suite-chatmodel-tck/src/main/java/io/micronaut/langchain4j/chatmodels/tck/tests/AiServiceTest.java
package io.micronaut.langchain4j.chatmodels.tck.tests;

import io.micronaut.context.annotation.Property;
import io.micronaut.context.annotation.Requires;
import io.micronaut.langchain4j.chatmodels.tck.SuiteCondition;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@Requires(condition = SuiteCondition.class)
@Property(name = "spec.name", value = "AiServiceTest")
@MicronautTest(startApplication = false)
public class AiServiceTest {
    @Test
    void testAiService(Friend friend) {
        String result = assertDoesNotThrow(() -> friend.chat("Hello"));
        Assertions.assertNotNull(result);
    }
}

File: test-suite-python/src/test/python/example/micronaut/aiservice/AiServiceTest.py
from typing import Annotated

from dev.langchain4j.model.chat import ChatModel
from jakarta.inject import Inject
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from example.micronaut.aiservice.Friend import Friend


@MicronautTest(startApplication=False, environments=["ollama"])
class AiServiceTest:
    friend: Annotated[Friend, Inject]
    language_model: Annotated[ChatModel, Inject]

    @Test
    def test_ai_service(self):
        result = self.friend.chat("Hello")

        assert result is not None
        assert self.language_model is not None
